n3fit/src/evolven3fit_new/evolve.py

```diff
@@ -90,28 +88,30 @@ def evolve_fit(
     if eko_path is not None:
         eko_path = pathlib.Path(eko_path)
         _logger.info(f"Loading eko from : {eko_path}")
-        eko_op = output.Output.load_tar(eko_path)
     else:
         try:
             _logger.info(f"Loading eko from theory {theoryID}")
-            theory_eko_path = (Loader().check_theoryID(theoryID).path)/'eko.tar'
-            eko_op = output.Output.load_tar(theory_eko_path)
+            eko_path = (Loader().check_theoryID(theoryID).path) / "eko.tar"
         except FileNotFoundError:
             _logger.info(f"eko not found in theory {theoryID}, we will construct it")
-            eko_op = eko_utils.construct_eko_for_fit(theory, op, dump_eko) 
-    eko_op.xgrid_reshape(targetgrid=x_grid, inputgrid=x_grid)
-    info = gen_info.create_info_file(theory, op, 1, info_update={})
-    info["NumMembers"] = "REPLACE_NREP"
-    info["ErrorType"] = "replicas"
-    info["AlphaS_Qs"] = list(eko_op["Q2grid"])
-    info["XMin"] = float(x_grid[0])
-    info["XMax"] = float(x_grid[-1])
-    dump_info_file(usr_path, info)
-    for replica in initial_PDFs_dict.keys():
-        evolved_block = evolve_exportgrid(initial_PDFs_dict[replica], eko_op, x_grid)
-        dump_evolved_replica(
-            evolved_block, usr_path, int(replica.removeprefix("replica_"))
-        )
+            runner.solve(theory, op, dump_eko)
+            eko_path = dump_eko
+    eko_op = eko.EKO.edit(eko_path)
+    with eko.EKO.edit(eko_path) as eko_op:
+        x_grid_obj = eko.interpolation.XGrid(x_grid)
+        eko.io.manipulate.xgrid_reshape(eko_op, targetgrid=x_grid_obj, inputgrid=x_grid_obj)
```

Comment:
Since you're manipulating only here, and then you're using it inside the loop, you consider closing here the `edit` context, and then reopen the EKO in read mode below.

Reply:
Yes this would be cleaner